### PR TITLE
Update to syn-2 (Continuation)

### PR DESCRIPTION
**Pull request by daxpedda**
Continuing where @maurer left off.

Do you need me to squash?

Replaces #352.

### CODE REVIEW

File: tests/ui/pin_project/add-attr-to-struct.rs [hunk 1]
```diff
@@ -4,7 +4,7 @@ use auxiliary_macro::add_pin_attr;
 use pin_project::pin_project;
 
 #[pin_project]
-#[add_pin_attr(struct)] //~ ERROR duplicate #[pin] attribute
+#[pin] //~ ERROR duplicate #[pin] attribute
```
Comment (daxpedda) [inline tests/ui/pin_project/add-attr-to-struct.rs:7]: Should I update comments like this to something more appropriate?

File: tests/ui/pin_project/add-attr-to-struct.stderr [hunk 2]
```diff
@@ -1,4 +1,4 @@
-error: duplicate #[pin] attribute
+error: expected attribute arguments in parentheses: `pin(...)`
```
Comment (daxpedda) [inline tests/ui/pin_project/add-attr-to-struct.stderr:1]: Should I update comments like this to something more appropriate?
https://github.com/taiki-e/pin-project/blob/ce860dd624aeabede645f38745aa9af334e559f0/tests/ui/pin_project/add-attr-to-struct.rs#L7

Comment (taiki-e) [reply tests/ui/pin_project/add-attr-to-struct.stderr:1]: Yes, please update these comments to match new error messages.

Comment (daxpedda) [reply tests/ui/pin_project/add-attr-to-struct.stderr:1]: Done.
I tried to keep the style consistent, let me know if you want me to change something.

**Review by taiki-e — APPROVED** [review]
Thanks!

**Comment by taiki-e** [comment]
Published in 1.1.0.

**Pull request merged**